### ISSUE TITLE
Open to more features?

### ISSUE THREAD

**Issue by stefanmaric** [pydsigner/python-lightningcss#4]
Hi @pydsigner  👋 

First of all, thanks for your contributions to the community through this package. ❤️ 

I needed to process some CSS on the fly in a Flask project and this module has been very helpful.

Since you have stated already that this project is "Opinionated" and "only a subset of functionality is exposed", I'm assuming you don't want to add additional features.

One feature I was missing was support for bundling, so I had to fork and publish a diff version: https://test.pypi.org/project/lightningcss-bindings/

I'm planning on expanding the bindings to be a more generic solution, but wanted to first check with you in case you prefer my contributions to be integrated in your project.

I've opened a pair of Pull Requests for chore tasks to give back to you:

https://github.com/pydsigner/python-lightningcss/pull/2
https://github.com/pydsigner/python-lightningcss/pull/3

Feel free to close them if you don't like it, I won't be mad!

Let me know your thoughts about collaborating on new features for `python-lightningcss`! I will wait a while before formalizing my fork and publishing to PyPi.

Thanks again!

**Comment by pydsigner** [comment]
Hi @stefanmaric, thanks for reaching out— I'm really glad to hear that you've found these bindings useful already! I did describe the bindings as opinionated, but that's more of a present tense thing, since there's just the one entry point and minimal flexibility. I would like to expose more functionality. #1 has been the big item in my perspective, since it would allow me to get rid of a bunch of arcane code in another project, but I'm open to items like bundling being exposed also.

**Comment by pydsigner** [comment]
I did take a look at your fork and noticed that you had dropped support for older Pythons. I would like to get further with the project before dropping any Python that hasn't at least reach EOL. (Please disregard if that was just to make builds run faster for testing...)

**Comment by stefanmaric** [comment]
> thanks for reaching out

Thanks for getting back!

> (Please disregard if that was just to make builds run faster for testing...)

I just talked about this in #3, it was definitely for testing 😄 

>  but I'm open to items like bundling being exposed also

@pydsigner so you have the time and energy to manage my contributions and publish new releases? - if so, I can prepare a PR to integrate the bundling function (which I'm actually using already via TestPyPi for the project I needed it for) and then work on more features (I was actually planning to write some integration tests soon to make the dev cycle shorter, to not have to publish to TestPyPi to verify the changes).

**Comment by pydsigner** [comment]
Yes I should be able to be fairly responsive on that— I've automated the release process enough that the pain of releases themselves is quite limited, and I have system tests in https://github.com/pydsigner/anchovy I can use for regression detection though tests are currently lacking here.

**Comment by pydsigner** [comment]
@mafr I never got a PR from @stefanmaric but the bundle CSS code he wrote is here: https://github.com/pydsigner/python-lightningcss/compare/master...stefanmaric:python-lightningcss:upgrades. It would need some adjustments to be mergeable but shouldn't be too hard to cherrypick into a PR if you're willing to take that on. I would certainly be willing to review and merge the change + make a release!

**Comment by pydsigner** [comment]
I would like to move to some sort of more modular structure to reduce how much code path duplication occurs but I'm aware that that's a longer-term goal and potentially much more work.

**Comment by mafr** [comment]
I have no Rust skills whatsoever, but I'll see if I can cherry-pick that commit and see if it works. Thank you!
